tests/test_csc.py

```diff
@@ -0,0 +1,302 @@
+
+            duration = 2  # seconds
+            task = asyncio.create_task(harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT+duration,
+                                                                           duration=duration))
+            await asyncio.sleep(0.1)  # give the event loop time to start
+            # the exposure state should be Integrating during the exposure
+            await self.check_exposureState(harness.remote, ExposureState.INTEGRATING)
+            await task
+            # the exposure state should be Done after a successful exposure
+            await self.check_exposureState(harness.remote, ExposureState.DONE)
+
+    async def test_expose_fails(self):
+        """Test that a failed exposure puts us in the FAULT state, which will
+        disconnect the device.
+        """
+        self.patch.return_value.AVS_GetScopeData.side_effect = None
+        self.patch.return_value.AVS_GetScopeData.return_value = AvsReturnCode.ERR_INVALID_MEAS_DATA.value
+        async with Harness(initial_state=salobj.State.ENABLED) as harness:
+            # Check that we are properly in ENABLED at the start
+            await self.check_summaryState(harness.remote, salobj.State.ENABLED)
+
+            msg = "Failed to take exposure"
+            with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED, result_contains=msg):
+                await harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT, duration=.5)
+            # the exposure state should be Integrating during the exposure
+            await self.check_exposureState(harness.remote, ExposureState.INTEGRATING)
+            await self.check_summaryState(harness.remote, salobj.State.FAULT)
+            error = await harness.remote.evt_errorCode.next(flush=False, timeout=STD_TIMEOUT)
+            errorMsg = str(AvsReturnError(AvsReturnCode.ERR_INVALID_MEAS_DATA.value, "GetScopeData"))
+            self.assertIn(errorMsg, error.errorReport)
+            # Going into FAULT should close the device connection.
+            self.assertIsNone(harness.csc.device)
+            # the exposure state should be FAILED after a failed exposure
+            await self.check_exposureState(harness.remote, ExposureState.FAILED)
+
+    async def test_cancelExposure(self):
+        """Test that we can stop an active exposure, and that the exposureState
+        is changed appropriately.
+        """
+        async with Harness(initial_state=salobj.State.ENABLED) as harness:
+            # Check that we are properly in ENABLED at the start
+            await self.check_summaryState(harness.remote, salobj.State.ENABLED)
+
+            duration = 5  # seconds
+            task = asyncio.create_task(harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT + duration,
+                                                                           duration=duration))
+            await asyncio.sleep(0.1)  # give the event loop time to start
```

Comment:
Consider instead calling `await self.check_exposureState(harness.remote, ExposureState.INTEGRATING)` here. That will tell you it has started integrating and it's safe to now cancel the exposure.